functions/lib/pubsub/processUploads.js

```diff
@@ -152,3 +158,40 @@ function compressImage(tempFilePath, imageManipulation) {
     return imageManipulation.compress(tempFilePath, 960);
   }
 }
+
+function readImageData(mediaPath, localFS) {
+  return function() {
+    const imageData = {}
+    return localFS.readFile(mediaPath)
+      .then((data) => {imageData.mediaData = data})
+      .then(readImageSize(mediaPath, localFS))
+      .then((data) => {imageData.mediaSize = data})
+      .then(()=> {
+        return imageData;
+      })
+      .catch((err)=>{
```

Comment:
No need for this `catch` since you're just re-throwing it, that would be the default behavior without it.